src/sentry/tasks/unmerge.py

```diff
@@ -249,6 +249,13 @@ def truncate_denormalizations(group):
         group_id=group.id,
     ).delete()
 
+    # XXX: This can cause a race condition with the ``FirstSeenEventCondition``
+    # where notifications can be erroneously sent if they occur in this group
+    # before the reprocessing of the denormalizated data completes, since a new
+    # ``GroupEnvironment`` will be created.
```

Comment:
I don't think there's any clean way to handle this without tracking it on a per-fingerprint basis, so just kind of ignoring this for now.